text/0000-bee-model-crate/0000-bee-model-crate.md

```diff
@@ -0,0 +1,334 @@
+Every transaction contains a mutable metadata which consists of following attributes:
+```rust
+pub struct TransactionMetadata {
+    pub transaction_hash: String,
+    pub is_solid: bool,
+    pub usize: snapshotIndex
+}
+```
+
+Even though Proof-Of-Work should be handled in its own RFC, this example shows how it could interact with the TransactionBuilder and how Transactions could be built.
+This example assumes that the actual Transaction building takes place here.
+
+
+```rust
+pub struct Pow;
+
+impl Pow {
+
+    // operates on String encoding, should be replaced with trits interface once ready
+    pub fn compute(transaction_builder: &TransactionBuilder) -> (Transaction, TranscationMetadata) {
+
+        const CHARSET: &[u8] = b"ABCDEFGHIJKLMNOPQRSTUVWXYZ9";
+
+        let iv: String = (0..27)
+            .map(|_| {
+                let mut rng = rand::thread_rng();
+                let idx = rng.gen_range(0, CHARSET.len());
+                char::from(unsafe { *CHARSET.get_unchecked(idx) })
+            })
+            .collect();
+
+        let trytes_backup = transaction_builder.signature_fragments.clone(); // + value + all other fields
+        let mut hash;
+        let mut nonce = iv.clone();
+        let mut timestamp = 0;
+        let mut transaction_metadata: TranscationMetadata = TranscationMetadata::default();
+
+        loop {
+
+            let mut trytes = trytes_backup.clone();
+            trytes.push_str(&nonce);
+
+            hash = HashFunction::hash(&trytes);
+            nonce = String::from(&hash[..27]);
+            // timestamp = ... (update timestamp)
+
+            if hash.ends_with(MIN_WEIGHT_MAGNITUDE_AS_STRING){
+                transaction_metadata.hash = hash.clone();
+                break
+            }
+
+        }
+
+        (
+            Transaction {
+                signature_fragments: transaction_builder.signature_fragments.clone(),
+                address: transaction_builder.address.clone(),
+                value: transaction_builder.value.clone(),
+                obsolete_tag: transaction_builder.obsolete_tag.clone(),
+                timestamp,
+                current_index: transaction_builder.current_index.clone(),
+                last_index: transaction_builder.last_index.clone(),
+                bundle_hash: transaction_builder.bundle_hash.clone(),
+                trunk: transaction_builder.trunk.clone(),
+                branch: transaction_builder.branch.clone(),
+                nonce,
+                tag: transaction_builder.tag.clone(),
+                attachment_timestamp: transaction_builder.attachment_timestamp.clone(),
+                attachment_timestamp_lower_bound: transaction_builder.attachment_timestamp_lower_bound.clone(),
+                attachment_timestamp_upper_bound: transaction_builder.attachment_timestamp_upper_bound.clone(),
+            },
+
+            transaction_metadata
+
+        )
+
+    }
+
+}
+
+```
+
+
+### Bundle
```

Comment:
This should go into its own RFC.

Reply:
Yeah not sure we  should mention PoW at all here.

Reply:
@SuperFluffy Agree